csharp-advanced-topics/SourceGeneratorInCSharp/Generator/ExampleGenerator.cs

```diff
@@ -0,0 +1,34 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Text;
+using System.Diagnostics;
+using System.Text;
+
+namespace Generator
+{
+    [Generator]
+    public sealed class ExampleGenerator : ISourceGenerator
+    {
+        public void Execute(GeneratorExecutionContext context)
+        {
+            var callingEntrypoint = context.Compilation.GetEntryPoint(context.CancellationToken);
+
+            var sourceText =
+    $@"namespace {callingEntrypoint.ContainingNamespace.Name}
+{{
+    public static class HelloWorld
+    {{
+        public static void SayHello()
+        {{
+            Console.WriteLine(""Hello From Generator"");
+        }}
+    }}
+}}";
```

Comment:
A raw string literal would make this cleaner and would remove the need to escape all the brackets and the quote marks. We can still use string interpolation within raw string literals.